sdk/python/feast/infra/materialization/snowflake_engine.py

```diff
@@ -276,27 +276,35 @@ def _materialize_one(
                     fv_latest_values_sql, feature_view.batch_source.field_mapping
                 )
 
-            fv_to_proto_sql = self.generate_snowflake_materialization_query(
-                self.repo_config,
-                fv_latest_mapped_values_sql,
-                feature_view,
-                project,
-            )
-
-            if self.repo_config.online_store.type == "snowflake.online":
-                self.materialize_to_snowflake_online_store(
+            features_full_list = feature_view.features
+            feature_batches = [
+                features_full_list[i : i + 100]
+                for i in range(0, len(features_full_list), 100)
+            ]
+            for feature_batch in feature_batches:
+
+                fv_to_proto_sql = self.generate_snowflake_materialization_query(
                     self.repo_config,
-                    fv_to_proto_sql,
+                    fv_latest_mapped_values_sql,
                     feature_view,
+                    feature_batch,
                     project,
                 )
-            else:
-                self.materialize_to_external_online_store(
-                    self.repo_config,
-                    fv_to_proto_sql,
-                    feature_view,
-                    tqdm_builder,
-                )
+
+                if self.repo_config.online_store.type == "snowflake.online":
+                    self.materialize_to_snowflake_online_store(
+                        self.repo_config,
+                        fv_to_proto_sql,
+                        feature_view,
+                        project,
+                    )
+                else:
+                    self.materialize_to_external_online_store(
```

Comment:
you might need to change how tqdm works here since it keeps on resetting as is for each feature batch, when you really probably want one big tqdm that increases partially for each batch of features